src/ploomber/spec/taskspec.py

```diff
@@ -46,6 +47,13 @@ def _looks_like_path(s):
         return '/' in s
 
 
+def _extension_typo(extension, valid_extensions):
+    try:
+        return get_close_matches(extension, valid_extensions)
```

Comment:
under which conditions this would throw an exception and what exception would that be? it's better to be specific (switch the `Exception` for the concrete exception it may raise)

Reply:
I realised it was throwing exception only when _safe_suffix returns None (no extension found) on product, e.g., product = ['raw', 'table']. So I have added a condition instead since it would be a faster operation.